### PR TITLE
 Core/Entities: fix excessive damage of exotic pets and some creatures 

### LINKED ISSUE #21333: Exotic hunter pet outputs excessive damage [60$]
**Description:**

The elite/rare pets of the "Exotic beasts" talent have excessive damage.

**Current behaviour:**

They cause damage from 10k to 20k depending on the player's equipment.

**Steps to reproduce the problem:**

1. Get the pet "Gondria"  (http://www.wowhead.com/npc=33776/gondria)
2. Equip http://www.wowhead.com/item-set=-273/wrathful-gladiators-pursuit y others pieces wrathful
3. Cause damage with the pet

**Branch(es):** 3.3.5

**TC rev. hash/commit:** 1acfefb973895b0e3df3c8d9b43b920d2a1b5392

**TDB version:** TDB 335.63

**Operating system:** Debian 9

### PR DESCRIPTION
**Pull request by ForesterDev**
**Changes proposed:**

-  Do not take into account secondary weapon damage for pets and creatures. This is the main reason of excessive damage 

**Target branch(es):** 3.3.5/master

- [x] 3.3.5

**Issues addressed:** Closes #21333  (insert issue tracker number)

**Tests performed:** (Does it build, tested in-game, etc.)
Tested on live server within months. Works fine

### CODE REVIEW

File: src/server/game/Entities/Unit/Unit.cpp [hunk 1]
```diff
@@ -1128,14 +1128,19 @@ void Unit::CalculateMeleeDamage(Unit* victim, CalcDamageInfo* damageInfo, Weapon
 
     for (uint8 i = 0; i < MAX_ITEM_PROTO_DAMAGES; ++i)
     {
+        // only players have secondary weapon damage
```
Comment (Aokromes) [inline src/server/game/Entities/Unit/Unit.cpp]: ain't some creatures supposed to use offhand weapon?

Comment (ForesterDev) [reply src/server/game/Entities/Unit/Unit.cpp]: This is not offhand calculation, this is the secondary effects of each damage types

Comment (ForesterDev) [reply src/server/game/Entities/Unit/Unit.cpp]: F.e. http://db.logon3.com/?item=1482 
+4 - 8 Shadow Damage
This is secondary weapon damage

**Comment by Killyana** [comment]
I tested it and it works fine, the damage is normalized.

**Comment by jackpoz** [comment]
Thanks for the PR :)

**Pull request merged**